### ISSUE TITLE
PC-lint report may create SonarException in decorator

### ISSUE THREAD

**Issue by Bertk** [wenns/sonar-cxx#600]
A rare scenario will create a java.lang.ArrayIndexOutOfBoundsException in decorator.
This will happen whenever a issue is detected after the complete file was processed e.g.
```number = "766" desc = "Header file 'some-header.h' not used in module 'src\mysource.cpp'"``` and there is no empty line at the end of the file. The line number for the issue in the PC-lint report file is incremented by 1 compared to the number of lines in the source file.

This was observed with SQ 5.1.1 and the C++ community plug-in.

Possible Solutions:
a) Add an empty line at the end of the source files
b) Correct the line number of the issue from the Report and set it to line=0 or the value for the last line.

**Comment by Bertk** [comment]
The SonarException will be observed for **RATS** and **PC-lint** reports. Both analysis tools create line numbers greater than the available *max line of the source file* and for me this looks like some "special feature". The C standard anyway demands a single empty line at the end of every file and I modified the Cxx plug-in to correct the report line number and limit it to the *max line of the source file* (4d4efdcb4b36c8e1dd04bb744c4a91b64a02ae94).
The exception is created only with SQ 5.x and not with the SQ 4.5.5 (LTS) version.

**Comment by Bertk** [comment]
see also SQ issue 6792 (https://jira.sonarsource.com/browse/SONAR-6792)
